### PR TITLE
Added Contributing.md

### PR DESCRIPTION
**Pull request by khalibartan**
I thought of adding a contribution guide in pgmpy.
Ping @ankurankan . Checkout the guide.
Tell me specific points you want to add apart from it or remove from it.

### CODE REVIEW

**Comment by kshitij10496** [comment]
Nice work @khalibartan ! 
+1

**Comment by khalibartan** [comment]
@kshitij10496 I think we should also work on documentation. [sympy](http://docs.sympy.org/latest/index.html) and [scikit-learn](http://scikit-learn.org/stable/documentation.html) really have a very good documentation. A good documentation really puts life into community.

**Comment by kshitij10496** [comment]
@khalibartan +1.
I was about to say that

**Comment by yashu-seth** [comment]
+1

File: Contributing.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  area) and with `git diff --cached` before commiting;
+- If you have push access to the main repository, please do not commit directly
+  to `dev`: your access should be used only to accept pull requests; if you
+  want to make a new feature, you should use the same process as other
+  developers so you code will be reviewed.
+
+
+### Code Guidelines
+
+- We use [PEP8](http://www.python.org/dev/peps/pep-0008/);
```
Comment (ankurankan) [inline Contributing.md:66]: @khalibartan Also do mention that we use 120 character line instead of 79 suggested by PEP8

**Comment by ankurankan** [comment]
@khalibartan Thanks a lot for this. We really needed this. Just do the minor change.

**Comment by khalibartan** [comment]
@ankurankan Fixed it.

File: Contributing.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+on how to report better issues.
+
+### Git and our Branching model
+
+#### Git
+
+We use [Git](http://git-scm.com/) as our [version control
+system](http://en.wikipedia.org/wiki/Revision_control), so the best way to
+contribute is to learn how to use it and put your changes on a Git repository.
+There's a plenty of documentation about Git -- you can start with the [Pro Git
```
Comment (yashu-seth) [inline Contributing.md]: @khalibartan There are plenty of documentations on Git --

File: Contributing.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+[gitflow](http://nvie.com/posts/a-successful-git-branching-model/) to manage our
+branches.
+
+Summary of our git branching model:
+- Fork the desired repository on GitHub to your account;
+- Clone your forked repository locally
+  (`git clone git@github.com:your-username:repository-name.git`);
+- Create a new branch off of `develop` with a descriptive name (for example:
+  `feature/portuguese-sentiment-analysis`, `hotfix/bug-on-downloader`). You can
+  do it switching to `develop` branch (`git checkout develop`) and then
```
Comment (yashu-seth) [inline Contributing.md]: @khalibartan do it by switching

File: Contributing.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+  creating a new branch (`git checkout -b name-of-the-new-branch`);
+- Do many small commits on that branch locally (`git add files-changed`,
+  `git commit -m "Add some change"`);
+- Push to your fork on GitHub (with the name as your local branch:
+  `git push origin branch-name`);
+- Create a pull request using the GitHub Web interface (asking us to pull the
+  changes from your new branch and add the to our `develop` branch);
```
Comment (yashu-seth) [inline Contributing.md]: @khalibartan ....and add the changes to our ......

File: Contributing.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+  messages](http://robots.thoughtbot.com/5-useful-tips-for-a-better-commit-message).
+- Anything in the `dev` branch should be deployable (no failing tests).
+- Never use `git add .`: it can add unwanted files;
+- Avoid using `git commit -a` unless you know what you're doing;
+- Check every change with `git diff` before adding then to the index (stage
+  area) and with `git diff --cached` before commiting;
+- If you have push access to the main repository, please do not commit directly
+  to `dev`: your access should be used only to accept pull requests; if you
+  want to make a new feature, you should use the same process as other
+  developers so you code will be reviewed.
```
Comment (yashu-seth) [inline Contributing.md]: @khalibartan so that your code can be reviewed.

**Comment by yashu-seth** [comment]
@khalibartan Just suggested few changes. If you also feel the same please make them.

**Comment by khalibartan** [comment]
@yashu-seth Thanks for pointing them out. I did the changes.

**Pull request merged**